dash.el

```diff
@@ -77,6 +77,14 @@ special values."
 
 (put '-each 'lisp-indent-function 1)
 
+(defalias '--each-indexed '--each)
+
+(defun -each-indexed (list fn)
+  "Call (FN index item) for each item in LIST.
+
+In the anaphoric form `--each-indexed', the index is exposed as `it-index`."
```

Comment:
Maybe we could also add a See also section here pointing to `-map-indexed` (and viceversa)?